### ISSUE TITLE
Scripture Reading Plans - The sequel

### ISSUE THREAD

**Issue by davidmoore1** [sillsdev/appbuilder-pwa#623]
Prerequisite: Issue #622 

Issue #622 dealt with most of the plan setup.  This issue mostly deals with how the plan affects the Scripture view, navigation, and completion and restart of a plan.

In the previous issue, it was discussed that if an reference is tapped from the calendar view of "My Plans", it goes to that reference.  One addition should be made here that it goes to that reference if the book defined in the reference is present in the current book collection.  If it isn't it does nothing.  

If the reference exists, then that reference becomes the active plan reference.  Note that while you can have multiple plans active at one time, you only have one active reference.  If you go to Plan 1 and select a reference but do not complete it, and then go to Plan 2 and select a reference, the reference for Plan 2 is the active plan reference and remains so until it is either closed, another plan reference is selected, or the app is restarted.  At app start, there is no active plan reference.  The active plan reference is independent of the book collection or layout settings.  So if you had cuk as the current book collection and selected Mark 1 as the reference from a plan, it remains the active plan reference if switch to the World English Bible version of Mark.  The plan reference is also not affected by any other type of navigation within the app.

One other thing worth mentioning in advance is that plan references cannot span chapters.  So if the plan for the day wants to include part or all of chapter 1 and all or part of chapter 2, there must be 2 references.  This is important because the part of the reference you are concerned with in the following sections will be the end verse of the active plan reference.  You do not need to worry about whether the end of the reference is in another chapter.  It is not.

When the verse at the end of that reference is displayed, an end of plan reference is displayed.  One thing to mention is that when this end of reference display is visible on the screen, the reference is marked as completed in the plan.  Even if the user does not press any button on the display, the entry is marked as complete.  If they view the plan calendar, this entry will now be checked.   It is still the "active plan reference" at this point until something else happens and if the user navigates away and comes back, this display will still be shown.  There are three forms of the end of reference display. 

The first is in the case where there are multiple references defined for this day in the plan and this is not the last entry.  The entry displays following the last verse of the reference indicating that the reference is completed, what the next reference is, a Next button and a close button.
![Simulator Screenshot - iPhone 15 - 2024-07-03 at 11 17 10](https://github.com/sillsdev/appbuilder-pwa/assets/6209188/f17fd091-901c-4bb6-bfd1-a41658b4bd1b)
If the entry is at the end of a chapter, it appears after the last verse and before any footer that may be configured:
![Simulator Screenshot - iPhone 15 - 2024-07-03 at 11 16 46](https://github.com/sillsdev/appbuilder-pwa/assets/6209188/bd10d96c-7cfb-4e72-a52a-353eb7cf1329)

The "x" close button causes the end of plan reference display to disappear.  It also clears the "active plan reference" state so that if this chapter is viewed again, the end of reference display will not appear.  The "Next" button will go to the next reference for the current plan day.  That entry now becomes the active plan reference. 

The second format is the case where the current entry is the last incomplete entry defined for the current plan day.  The entry displays following the last verse of the reference indicating that the reference is completed, a "View Plan" button and a close button.  
![Simulator Screenshot - iPhone 15 - 2024-07-03 at 11 17 39](https://github.com/sillsdev/appbuilder-pwa/assets/6209188/8823c3d6-eecf-4ca4-9354-2aa540882352)
The close button functions the same as the one for the first version.  The "View Plan" button takes you back to the calendar tab of the plan detail view for the plan associated with the active reference: 
![Simulator Screenshot - iPhone 15 - 2024-07-03 at 11 51 30](https://github.com/sillsdev/appbuilder-pwa/assets/6209188/e1d3ea97-be05-4261-aeb0-33f2e3f04024)
Pressing "View Plan" does not clear the active plan reference.  If you navigate back to that reference and the close button hasn't been pressed, the end of reference marker will still display until another reference is chosen from the plan.

The third format is the case where the current entry is the last uncompleted entry.  As mentioned earlier, when these displays become visible, the reference is marked as completed.  When this entry displays, the entry is marked completed and the plan is marked as completed.  All progress information is deleted from the database.  This view contains a congratulations message, a close button and a "View Plans" button.  If the "x" close button is pressed, the view disappears and the active plan reference is cleared.  If the "View Plans" button is pressed, the list of active tabs plan is displayed.
![Simulator Screenshot - iPhone 15 - 2024-07-03 at 10 25 55](https://github.com/sillsdev/appbuilder-pwa/assets/6209188/6b7561e4-409f-4731-b9e4-c7c46820e365)

 The last entry for plan and last entry for plan day were described in the previous issue.

Now we will discuss the "Completed" tab of the main "Plans" menu screen.  The "Completed" tab appears only when there are plans that have been completed that have not been restarted.  It shows all the plans with a last status of completed.  All plans that appear in this tab also appear in the "Choose Plan" tab.
![Simulator Screenshot - iPhone 15 - 2024-07-03 at 10 26 07](https://github.com/sillsdev/appbuilder-pwa/assets/6209188/a152d93f-1cf1-436f-ae1e-8cf9d384fd33)

If the user taps on a plan in the "Completed" tab, it displays a detail tab similar to the one when a plan is selected from "Choose Plan".  Two tabs display: one for the information screen which now also shows the date the plan was completed.  It also has a "Start Again" button.  If this button is tapped, the status of the project is changed to in progress and the calendar screen for this plan in the "My Plans" tab is shown.  It will no longer appear in the "Completed" plans tab.
![Simulator Screenshot - iPhone 15 - 2024-07-03 at 10 26 19](https://github.com/sillsdev/appbuilder-pwa/assets/6209188/aad58809-2395-42a9-95b3-112928995ff5)

If the calendar tab is displayed from the preview, then the calendar is displayed.  All entries are marked with a checkmark since they are all completed.  Pressing an entry on the calendar does nothing.  You can't get to a reference by pushing on the entry for a completed plan.
![Simulator Screenshot - iPhone 15 - 2024-07-03 at 10 26 36](https://github.com/sillsdev/appbuilder-pwa/assets/6209188/5fdd1d33-45a9-4013-b442-8cf21d295f25)

**Comment by chrisvire** [comment]
Completed by @davidmoore1 .
